#!/usr/bin/env python
PACKAGE = "saliency_detector"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("use_SVM_classifier", bool_t, 0, "Classify using trained SVM", False)
gen.add("min_perimeter_ratio", double_t, 0, "Minimum perimeter ratio", 0.7, 0.0, 1.0)
gen.add("max_perimeter_ratio", double_t, 0, "Maximem perimeter ratio", 1.0, 0.0, 1.0)
gen.add("min_area_ratio", double_t, 0, "Minimum area ratio", 0.7, 0.0, 1.0)
gen.add("max_area_ratio", double_t, 0, "Maximem area ratio", 1.0, 0.0, 1.0)
gen.add("min_defect_ratio", double_t, 0, "Minimum defect ratio", 0.7, 0.0, 1.0)
gen.add("max_defect_ratio", double_t, 0, "Maximem defect ratio", 1.0, 0.0, 1.0)

gen.add("window_size", double_t, 0, "Close window image half-width", 200, 100, 400)

exit(gen.generate(PACKAGE, "saliency_detector", "close_range_detector_params"))
